### ISSUE TITLE
Distillation on small data

### ISSUE THREAD

**Issue by baochi0212** [NovaSearch-Team/RAG-Retrieval#91]
Hi @NLPJCL, I try to sanity check stella distillation on arguana dataset from BGE en icl to bge-large-en v1.5 but encouter NaN loss 
Here is my steps:
- Create distillation data (Using arguana datasets from huggingface)
python create_distill_data.py ./distill_datasets/arguana ./bge-en-icl ./test.mmap 0  
- Train embedding:

![Image](https://github.com/user-attachments/assets/039d16b2-8b7f-42d0-bc55-9d7c02d4a969)
- Config

![Image](https://github.com/user-attachments/assets/128b04b0-1fb7-41a5-ba1e-4de197e207c6)

**Comment by baochi0212** [comment]
and I work around but add accelerator to DistillEmbedding forward

**Comment by NLPJCL** [comment]
You can provide the training log, the part just started.

**Comment by baochi0212** [comment]
Here's the log, thanks @NLPJCL 
![Image](https://github.com/user-attachments/assets/cd72ad6c-8e57-4e26-8cbb-a428090185fd)

**Comment by NLPJCL** [comment]
It seems that the first loss is normal, while the second and third losses are not normal. It is recommended that you
1. Check whether the text and vector in the training data correspond one to one?
2. Remove the second and third losses in the code to see if they converge normally.
3. Set mrl_dims: "256,512,1024,1536,2048,2560,3072,3584" to the default in config.

**Comment by baochi0212** [comment]
![Image](https://github.com/user-attachments/assets/00f3bb06-841f-4504-bd6c-1e63265ebb36)
The loss seems good but evaluation results still not work (recall = 0), hmm but when I train with "train" mode the results normal

**Comment by NLPJCL** [comment]
hi，you must set use_mrl: true, The last two losses will be useful.
When the amount of data is relatively small, it is normal that distillation fails.

**Comment by baochi0212** [comment]
Hi @NLPJCL, seems like the loss ok now, add MRL both helps, but seems like embedding level rep distillation not works for small dataset. tks for your help, I'll try on more large-scale one

**Comment by baochi0212** [comment]
yeah, add use_mrl helps, but the results degraded

**Comment by NLPJCL** [comment]
Generally, distillation requires at least hundreds of thousands to millions of data points.

**Comment by baochi0212** [comment]
yeah, just to check my intuition, tks for fast response :D
